haystack/nodes/_json_schema.py

```diff
@@ -156,9 +156,20 @@ def handle_optional_params(param_fields: List[inspect.Parameter], params_schema:
         else:
             anyof_list = param_dict.pop("anyOf", None)
             if anyof_list is not None:
+                empty_items = []
```

Comment:
@ZanSara do we need these changes here? I'm asking because I am not the most familiar with this schema :-)

Reply:
I'm also curious why this was necessary. There might be a good reason but from the diff it's really hard to tell

Reply:
if we define a parameter as following in a __init__ function of component node which can be used in pipeline yaml:
invocation_layer_class: Optional[Union[Type[PromptModelInvocationLayer], str]] = None

The pydantic generates following schema:
schema= {'invocation_layer_class': {'title': 'Invocation Layer Class', 'anyOf': [{}, {'type': 'string'}]}}

The empty {} will cause the exception of following line:
anyof_list = list(sorted(anyof_list, key=lambda x: x["type"]))